src/nvim/main.c

```diff
@@ -1703,6 +1705,20 @@ static bool do_user_initialization(void)
     do_exrc = p_exrc;
     return do_exrc;
   }
+  char_u *init_lua_path = (char_u *)stdpaths_user_conf_subpath("init.lua");
+
+  if (execute_init_lua(init_lua_path)) {
+    // the 'exrc' script should also be executed
+    do_exrc = p_exrc;
+    if (do_exrc) {
+      do_exrc = (path_full_compare((char_u *)VIMRC_FILE, init_lua_path, false)
+                 != kEqualFiles);
+    }
+    xfree(init_lua_path);
+    return do_exrc;
+  }
+  xfree(init_lua_path);
+
   char_u *user_vimrc = (char_u *)stdpaths_user_conf_subpath("init.vim");
```

Comment:
It is an error if init.vim _and_ init.lua exists. That avoids "ordering" decisions and keeps the documentation simpler (and mental model for users).